tests/test_benchmark.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from os import path
+from conftest import skipif
+from devito import configuration
+from subprocess import check_call
+
+
+pytestmark = skipif(['yask', 'ops'])
+
+
+@pytest.mark.parametrize('mode', ['bench'])
+@pytest.mark.parametrize('problem', ['acoustic', 'tti', 'elastic', 'viscoelastic'])
+def test_bench(mode, problem):
+    """
+    Test the Devito benchmark framework on various combinations of modes and problems.
+    """
+    if configuration['openmp']:
+        nthreads = configuration['platform'].cores_physical
+    else:
+        nthreads = 1
+
+    command_bench = ['python', 'benchmarks/user/benchmark.py', mode, '-P', problem,
+                     '-d', '16', '16', '16', '--tn', '4', '-x', '1']
+    check_call(command_bench)
+
+    dir_name = 'results/'
+
+    base_filename = problem
+    filename_suffix = '.json'
+    arch = 'arch[unknown]'
+    shape = 'shape[16,16,16]'
+    nbl = 'nbl[10]'
+    tn = 'tn[4]'
+    so = 'so[2]'
```

Comment:
@mloubout `so[2]` and TTI running? 😮

Reply:
Yeah it runs, not recommended for the quality of solution but runs fine